subsys/net/lib/nrf_cloud/src/nrf_cloud_pgps.c

```diff
@@ -1383,13 +1390,70 @@ static int consume_pgps_data(uint8_t pnum, const char *buf, size_t buf_len)
 	return 0;
 }
 
+#if !defined(CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE)
+static
+#endif
+int nrf_cloud_pgps_begin_update(void)
+{
+	int err;
+
+#if defined(CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE)
```

Comment:
This is just a draft.  I did this in a hurry on behalf of the LWM2M team, but plan to clean it up before turning off draft.

Reply:
Makes sense, I apologize if the above suggestion was obvious / something you were already planning to do

Reply:
I realized today that actually, there are two separate features to configure -- the transport for the P-GPS request and cloud response, and the transport for the P-GPS prediction data itself.

There are multiple possible combinations of these:

- MQTT req/resp transport, HTTP(S) prediction transport -- CONFIG_NRF_CLOUD_PGPS_TRANSPORT_MQTT + CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_TRANSPORT_HTTP
- REST req/resp transport, HTTP(S) prediction transport
CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE + CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_TRANSPORT_HTTP
- LWM2M req/prediction transport (resp handled by LWM2M server, AVSystem)
CONFIG_NRF_CLOUD_PGPS_TRANSPORT_NONE + CONFIG_NRF_CLOUD_PGPS_DOWNLOAD_TRANSPORT_CUSTOM

Reply:
In my opinion (without looking at your latest code, on to that in a minute), the request transport should be named `CONFIG_PGPS_REQUEST_TRANSPORT`, and should have the following values:

- `CONFIG_PGPS_REQUEST_TRANSPORT_NRF_CLOUD_MQTT`
- `CONFIG_PGPS_REQUEST_TRANSPORT_NRF_CLOUD_REST`
- `CONFIG_PGPS_REQUEST_TRANSPORT_CUSTOM`

With the AVSystem stuff either using CUSTOM, or a fourth value, `CONFIG_PGPS_REQUEST_TRANSPORT_AVSYSTEM_LWM2M`, depending on what makes more sense.

Similarly, I think the download transport should be named `CONFIG_PGPS_DOWNLOAD_TRANSPORT` and have the values:
- `CONFIG_PGPS_DOWNLOAD_TRANSPORT_HTTPS`
- `CONFIG_PGPS_DOWNLOAD_TRANSPORT_CUSTOM`

Custom in each case just indicates that an external library will execute certain calls, or register appropriate callbacks, to hook into the system in a customized manner.

`CONFIG_PGPS_REQUEST_TRANSPORT_NRF_CLOUD_MQTT` and `CONFIG_PGPS_REQUEST_TRANSPORT_NRF_CLOUD_REST` should probably `depends on PGPS_DOWNLOAD_TRANSPORT_HTTP`

Reply:
I hesitate to rename the older configs due to the need to then touch a bunch of other places in NCS (samples, the location library, etc.).

REST is problematic, because the actual use of REST is outside of this library -- it is provided by the modules that use P-GPS with REST, like the location library, the gnss sample, etc.  I would have to also refactor all of those places too.  Otherwise, it's misleading to say "CONFIG_PGPS_REQUEST_TRANSPORT_NRF_CLOUD_REST" if the REST part is not built in as the MQTT part is.

Such a refactoring could be done, but I think that is best done as another step, and perhaps after the 2.1.0 release.

Reply:
I think in the long term a refactoring like that would be worthwhile, but you're right, maybe not for this release or for this PR. 

I didn't realize that the libraries performing PGPS requests over REST were essentially doing so in the form of a "custom transport." IMO, we should move toward eventually integrating the REST option directly into the library

But again, you're right, that might be a task for another PR